packages/terminal/src/browser/base/terminal-widget.ts

```diff
@@ -25,6 +25,11 @@ export abstract class TerminalWidget extends BaseWidget {
 
     abstract processId: Promise<number>;
 
+    /**
+     * Whether the terminal widget runs a task
+     */
+    abstract hasRunningTask: boolean;
```

Comment:
please revert, terminal should not know anything about tasks

Reply:
could you please suggest another way to check if the terminal is idle?
or maybe i can simply rename `hasRunningTask` to `isIdle` to make it look irrelevant to tasks?

Reply:
How it would be useful if we forget about tasks? There should be conceptual explanation.

Reply:
Yes `isIdle` is conceptually irrelevant to tasks.
in the terminal widget class `isIdle` is set `true` once the attached process finishes.

Reply:
I mean what does it mean?